src/main/scala/net/snowflake/spark/snowflake/LoggerWrapper.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2015-2020 Snowflake Computing
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.snowflake.spark.snowflake
+
+import java.io.OutputStream
+
+import org.slf4j.helpers.MessageFormatter
+import org.slf4j.{Logger, LoggerFactory}
+
+/**
+  * Logger Wrapper class.
+  *
+  * The logger wrapper may do some special handling for the log message,
+  * for example, sending telemetry messages. And then pass through the logging
+  * request to internal logger object.
+  *
+  * NOTE: LoggerWrapper only implements the functions which are used by Spark
+  *       Connector. If other functions in <code>org.slf4j.Logger</code> need
+  *       to be used, they need to be implemented in this class.
+  */
+private[snowflake] class LoggerWrapper(logger: Logger) {
+
+  private var isInstanceSendLogTelemetryEnabled: Boolean = true
+
+  private[snowflake] def enableInstanceSendLogTelemetry(): Unit = {
+    isInstanceSendLogTelemetryEnabled = true
+  }
+
+  private[snowflake] def disableInstanceSendLogTelemetry(): Unit = {
+    isInstanceSendLogTelemetryEnabled = false
+  }
+
+  // Below variables and functions are for test purpose
+  private var enableTest = false
+  private var testOutputStream: OutputStream = _
+  private[snowflake] def enableTest(outputStream: OutputStream): Unit = {
```

Comment:
My preference would be to refactor this to make this a test LoggerWrapper class implement LoggerWrapper instead of using this enableTest thing.

Reply:
Good idea. I have decoupled TelemetryReporter from LoggerWrapper. I can add a TestTelemetryReporter for this. I will update it.